djsonb/lookups.py

```diff
@@ -72,10 +74,27 @@ def sql(self):
                         sql_tuple = FilterTree.text_similarity_filter(rule[0], pattern, True)
                     else:
                         sql_tuple = FilterTree.text_similarity_filter(rule[0], pattern, False)
-                    rule_specs.append(sql_tuple)
+                    # add to the list of rules generated for this pattern (one per field)
+                    got_pattern = patterns.get(pattern) or []
+                    got_pattern.append(sql_tuple)
+                    patterns[pattern] = got_pattern
+
+        rule_strings = [' AND '.join([rule[0] for rule in rule_specs])]
+
+        pattern_rules = patterns.values()
+        pattern_strings = []
+
+        for rule_list in pattern_rules:
+            pattern_strings.append(' OR '.join([rule[0] for rule in rule_list]))
+            pattern_specs += rule_list
+
+        if pattern_strings:
+            pattern_string = '(' + ') AND ('.join(pattern_strings) + ')'
+        else:
+            pattern_string = ''
```

Comment:
All this extra conditionality just to get enough but no too many sets of parens is a drag, but I don't see a nice way to fix it.